lib/puppet-lint/plugins/check_parameter_documentation.rb

```diff
@@ -25,51 +23,76 @@ def check
         end
       end
 
-      next if is_private
-
-      params = []
-      e = idx[:param_tokens].each
-      begin
-        while (ptok = e.next)
-          if ptok.type == :VARIABLE
-            params << ptok
-            nesting = 0
-            # skip to the next parameter to avoid finding default values of variables
-            while true
-              ptok = e.next
-              case ptok.type
-              when :LPAREN
-                nesting += 1
-              when :RPAREN
-                nesting -= 1
-              when :COMMA
-                break unless nesting > 0
-              end
-            end
-          end
-        end
-      rescue StopIteration; end
+      params = extract_params(idx)
 
       # warn about duplicates
       doc_params_duplicates.each do |parameter, tokens|
         tokens.each do |token|
            notify :warning, {
-             :message => "Duplicate parameter documentation for #{parameter}",
+             :message => "Duplicate #{type_str(idx)} parameter documentation for #{parameter}",
              :line    => token.line,
              :column  => token.column + token.value.match(/\A\s*(@param\s*)?/)[0].length + 1 # `+ 1` is to account for length of the `#` COMMENT token.
            }
         end
       end
 
+      # warn about documentation for parameters that don't exist
+      doc_params.each do |parameter, token|
+        next if params.find { |p| p.value == parameter }
+
+        notify :warning, {
+          :message => "No matching #{type_str(idx)} parameter for documentation of #{parameter}",
+          :line    => token.line,
+          :column  => token.column + token.value.match(/\A\s*(@param\s*)?/)[0].length + 1
+        }
+      end
+
+      # Warn about missing documentation unless class/type is marked private
+      next if is_private
```

Comment:
Personally I never like `next` and would prefer to have something like `unless is_private ...`.

Reply:
Fixed